py-polars/polars/plugins.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from polars._utils.parse_expr_input import parse_as_list_of_expressions
+from polars._utils.udfs import get_dynamic_lib_location
+from polars._utils.unstable import unstable
+from polars._utils.wrap import wrap_expr
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from polars import Expr
+    from polars.type_aliases import IntoExpr
+
+__all__ = ["register_plugin"]
+
+
+@unstable()
+def register_plugin(
+    *inputs: IntoExpr | list[IntoExpr],
+    plugin_location: str | Path,
+    symbol: str,
+    kwargs: dict[Any, Any] | None = None,
+    is_elementwise: bool = False,
+    input_wildcard_expansion: bool = False,
+    returns_scalar: bool = False,
+    cast_to_supertypes: bool = False,
+    pass_name_to_apply: bool = False,
+    changes_length: bool = False,
+) -> Expr:
+    """
+    Register a dynamic library as a plugin.
+
+    .. warning::
+        This is highly unsafe as this will call the C function
+        loaded by `lib::symbol`.
+
+        The parameters you give dictate how polars will deal
+        with the function. Make sure they are correct!
+
+    .. note::
+        This functionality is unstable and may change without it
+        being considered breaking.
```

Comment:
Can we use the default warning here? And move it to right below the docstring summary (e.g. above the other warning)?

```
    .. warning::
        This functionality is considered **unstable**. It may be changed
        at any point without it being considered a breaking change.
```